client/disputes/utils.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import { CachedDispute, EvidenceDetails } from 'wcpay/types/disputes';
+
+/**
+ * Returns true if a dispute due_by date is within the specified number of days.
+ * Returns false if the dispute due_by date is not within the specified number of days
+ * or if the due_by value is not a valid date.
+ *
+ * @param {Object} dispute - The dispute to check. See {@link CachedDispute} or {@link EvidenceDetails}.
+ * @param {number} days - The number of days to check.
+ *
+ * @return {boolean} True if the dispute is due within the specified number of days.
+ */
+interface IsDueWithinProps {
+	dueBy: CachedDispute[ 'due_by' ] | EvidenceDetails[ 'due_by' ];
+	days: number;
+}
+export const isDueWithin = ( { dueBy, days }: IsDueWithinProps ): boolean => {
```

Comment:
Ideally, split the creation of this shared util function out into a separate PR to aide reviewing and testing.